vmware/build-tools-for-vmware-aria
# Support for binding workflow attributes to config element values

### Description

This Pull Request is enabling the following feature for Aria Automation Orchestrator Workflows

*Be able to bind attribute values to Configuration Element variables*

To do that one have to provide the following annotation for the workflow ts file (SomeFile.wf.ts)

```
@Workflow({
        id: "<Some Id>",
        name: "<Some Name>",
        path: "<Some Path>",
        attributes: {
            attributeName: {
            type: "string",
            bind: true,   
            value: "Some/Path/To/ConfigurationElement/variableName"
            }
        }
    })
```
Here 
`bind: true` - means that we have to bind the value of the attribute to Configuration Element variable.
`value: "Some/Path/To/ConfigurationElement/variableName"` - points to the Configuration Element and variable inside the Configuration Element to bind to.

### Checklist

<!--


- [X] I have added relevant error handling and logging messages to help troubleshooting
- [X] I have updated CHANGELOG.md with a short summary of the changes introduced
- [X] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)

### Testing

It was tested in the following way. 
- Compile a SNAPSHOT version of built tools for VMware Aria for vrotsc
- Take a real VRO project which contains workflow definitions in Typescript format (*.wf.ts) and make it point (in pom.xml) to the SNAPSHOT version of Build tools for VMware Aria
- Extend the workflow in the VRO project with a Configuration Element and a workflow definition with attribute that is boud to a variable in the configuraiton element.
- Build the VRO Project to generate a `*.package` file 
- Import the `*.package` file in a live instance of Aria Automation Orchestrator and make sure that:
  * It imports successfully
  * The Workflow that is imported has an attribute definition.
  * The attribute definition is bound to the Configuration Element variable (which is also imported)

### Release Notes

Allow definition of workflow attribute which value is bound to the value of a Configuration Element variable.

### Related issues and PRs
